### PR TITLE
First implementation of wasm doc generation

### PR DESCRIPTION
**Pull request by TheQuantumPhysicist**
Question is open whether I should include parameters

### CODE REVIEW

File: wasm-wrappers/wasm-doc-gen/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+        })
+        .filter_map(|v| match &v.value {
+            syn::Expr::Lit(lit) => Some(lit),
+            _ => None,
+        })
+        .filter_map(|l| match &l.lit {
+            syn::Lit::Str(s) => Some(s),
+            _ => None,
+        })
+        .map(|s| s.value().trim().to_string())
```
Comment (iljakuklic) [inline wasm-wrappers/wasm-doc-gen/src/main.rs:191]: All of this could be just one `filter_map`. Not sure how much improvement would it be though.

**Review by iljakuklic — APPROVED** [review]
Looks good to me. Just some minor comments.

File: wasm-wrappers/wasm-doc-gen/src/main.rs (hunk 1 above)
Comment (iljakuklic) [reply wasm-wrappers/wasm-doc-gen/src/main.rs:191]: Something like (just typing here, not guaranteed to work):
```rust
.filter_map(|m| {
    let m = m.meta.require_name_value().ok()?;
    m.path().is_ident("doc").then_some(())?;  // poor man's `ensure!`
    let val = &m.value;
    let str_lit = quote!(#val).parse2::<syn::LitStr>().ok()?;
    Some(str_lit.value().trim().to_string())
})
```

File: wasm-wrappers/wasm-doc-gen/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+    }
+}
+
+fn pull_docs_from_attribute(attrs: &[syn::Attribute]) -> String {
+    let docs = attrs
+        .iter()
+        .filter_map(|m| m.meta.require_name_value().ok())
+        .filter(|m| {
+            m.path.segments.first().is_some()
+                && m.path.segments.first().expect("Was checked").ident == "doc"
```
[the remark this answers is not in the document]
Comment (TheQuantumPhysicist) [reply wasm-wrappers/wasm-doc-gen/src/main.rs:181]: Oh, thanks. Nice.

**Review by iljakuklic — APPROVED** [review]

**Pull request merged**